lib/babe/build.go

```diff
@@ -229,8 +229,7 @@ func (b *BlockBuilder) buildBlockExtrinsics(slot Slot, rt ExtrinsicHandler) []*t
 func buildBlockInherents(slot Slot, rt ExtrinsicHandler, parent *types.Header) ([][]byte, error) {
 	// Setup inherents: add timstap0
 	idata := types.NewInherentData()
-	timestamp := uint64(time.Now().UnixMilli())
-	err := idata.SetInherent(types.Timstap0, timestamp)
+	err := idata.SetInherent(types.Timstap0, uint64(slot.start.UnixMilli()))
```

Comment:
Shouldn't this be current time stamp? I have asked in the elements chat for confirmation.

Reply:
Looks like @EclesioMeloJunior also thinks that it should be current timestamp

Reply:
oh really? Okay will update this

Reply:
So to be clear, by "current timestamp" they meant current as in when the inherent are created as opposed to current being when the slot is claimed? @kishansagathiya @EclesioMeloJunior

Reply:
I think we do this in other parts of the code as well https://github.com/ChainSafe/gossamer/blob/2d5ead18ed3aa8fd1e412b26755a165de2248c02/dot/sync/chain_processor_integration_test.go#L46

Reply:
Talked with @EclesioMeloJunior and I understand now, will update!

Reply:
@kishansagathiya Upon further investigation, the slot.start value is set when the slot is activated, so the time will always be within the appropriate slot. So I think leaving it as is is okay since it allows easier testing and is still producing a correct timestamp.

https://github.com/ChainSafe/gossamer/blob/4ea0a703339ca06e617b06eeb7d56308e7b48093/lib/babe/babe.go#L485

Reply:
> it allows easier testing 

Is there a test or a usecase that fails, if you use `time.Now()`

> is still producing a correct timestamp.

Sort of. slot duration is generally in seconds (right?). timestamp's accuracy is a lot more than that.

> the slot.start value is set when the slot is activated, so the time will always be within the appropriate slot

That's kinda what I have  a problem with. I don't think it's a good idea to forcefully match slot and timestamp. They should match naturally. Maybe, using `slot.start` is good enough and doesn't create any problems. But, we know that that is not what is being expected from the spec. And I don't know what kind of problems it might lead.

This change looks like a hack to (may be) make things easier for now. But unless there is a strong need for it, we shouldn't have such hacks in our codebase.
@timwu20